crypto-core/src/main/java/com/palantir/crypto2/cipher/SeekableCipher.java

```diff
@@ -18,33 +18,47 @@
 
 import com.palantir.crypto2.keys.KeyMaterial;
 import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import org.apache.commons.crypto.cipher.CryptoCipher;
 
 /**
- * Provides access to a {@link Cipher} with the ability to {@link #seek} the Cipher.
+ * Provides access to a {@link CryptoCipher} with the ability to {@link #seek} the Cipher.
  * // TODO(jellis): add note about using SeekableCipher streams instead of this directly
  */
 public interface SeekableCipher {
 
     /**
-     * Initialize the underlying {@link Cipher} to either {@link Cipher#ENCRYPT_MODE} or {@link Cipher#DECRYPT_MODE}.
+     * Initialize the underlying {@link CryptoCipher} to either {@link Cipher#ENCRYPT_MODE} or
+     * {@link Cipher#DECRYPT_MODE}.
      */
-    Cipher initCipher(int opmode);
+    CryptoCipher initCipher(int opmode);
 
     /**
-     * The returned {@link Cipher} is initialized such that future operations will encrypt/decrypt correctly for the
-     * given byte offset {@code pos} into the plaintext data. Certain Ciphers have special requirements and restrictions
-     * on how and where they are able to be seeked to.
+     * The returned {@link CryptoCipher} is initialized such that future operations will encrypt/decrypt correctly for
+     * the given byte offset {@code pos} into the plaintext data. Certain ciphers have special requirements and
+     * restrictions on how and to where they are able to be seeked.
      */
-    Cipher seek(long pos);
+    CryptoCipher seek(long pos);
 
     /**
      * Returns the {@link KeyMaterial} being used by this {@link SeekableCipher} for cryptographic operations.
      */
     KeyMaterial getKeyMaterial();
 
     /**
-     * Returns the underlying {@link Cipher}'s block size.
+     * Returns the underlying {@link CryptoCipher}'s block size.
      */
     int getBlockSize();
 
+    /**
+     * Returns the algorithm used by the underlying {@link CryptoCipher} for encryption and decryption.
+     */
+    String getAlgorithm();
+
+    /**
+     * Returns the initialization vector for the block that the underlying {@link CryptoCipher} will encrypt or decrypt
+     * next.
+     */
+    IvParameterSpec getCurrIv();
```

Comment:
It looks like a `CipherInputStream` requires that the given `Cipher` be initialized already: http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/javax/crypto/CipherInputStream.java?av=f#28.

However, a `CryptoInputStream` will actually initialize the given `CryptoCipher` for you using information that you give it, which includes the parameter spec: https://github.com/apache/commons-crypto/blob/master/src/main/java/org/apache/commons/crypto/stream/CryptoInputStream.java#L488

So I'm glad you asked, because now it occurs to me that we're probably doing extra work by initializing the `CryptoCipher` before calling `CipherStreamSupplier#getInputStream` and `CipherStreamSupplier#getOutputStream`. I'll see if we can remove some cipher initialization from the diff here.